src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { ChangeEvent, useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { getFilteredTodos } from './utils/getFilteredTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterType>(FilterType.all);
+  const [query, setQuery] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const handleChangeFilter = (event: ChangeEvent<HTMLSelectElement>) => {
+    setFilter(event.target.value as FilterType);
+  };
+
+  const handleQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const queryCleaner = () => {
+    setQuery('');
+  };
+
+  const handleSelectTodo = (todo: Todo) => {
+    setSelectedTodo(todo);
+  };
+
+  const handleClearSelectedTodo = () => {
+    setSelectedTodo(null);
+  };
+
+  const filteredTodos = useMemo(() => {
+    const filtered = getFilteredTodos(todos, filter);
+
+    if (query) {
+      return filtered.filter(todo =>
+        todo.title.toLowerCase().includes(query.toLowerCase().trimStart()),
+      );
+    }
+
+    return filtered;
+  }, [filter, query, todos]);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .then(() => setLoading(false));
```

Comment:
```suggestion
      .finally(() => setLoading(false));
```

It's better to use finally in this case.